on_device_training/desktop/python/mnist.ipynb

```diff
@@ -197,6 +200,35 @@
     "onnx.save(eval_model, \"data/eval_model.onnx\")"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Method 2:** \n",
+    "\n",
+    "Alternatively, you can use the generate_artifacts function provided by the onnxblock library. This function automatically generates a training graph based on the forward-only graph and the specified loss function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "requires_grad = [name for name, param in pt_model.named_parameters() if param.requires_grad]\n",
+    "\n",
+    "frozen_params = [name for name, param in pt_model.named_parameters() if not param.requires_grad]\n",
+    "\n",
+    "artifacts.generate_artifacts(\n",
```

Comment:
From the first method, it seems like the graph has two outputs. From the second, it will not generate two outputs.

Since the rest of the training relies on there being two outputs, it might fail. Were you able to try it out?